crates/typst-cli/src/args.rs

```diff
@@ -132,8 +133,53 @@ pub struct SharedArgs {
         value_parser = clap::value_parser!(DiagnosticFormat)
     )]
     pub diagnostic_format: DiagnosticFormat,
+
+    /// Add an input value visible through sys.inputs
+    /// whose type is always treated as a string
+    /// which you may enclose with single or double quotes
+    #[clap(
+        long = "input",
+        short = 'i',
+        value_name = "KEY=VALUE",
+        action = ArgAction::Append,
+        value_parser = ValueParser::new(parse_input_pair),
+    )]
+    pub plain_inputs: Vec<(String, String)>,
 }
 
+fn parse_input_pair(raw: &str) -> Result<(String, String), InputPairError> {
+    let (key, val) = raw.split_once('=').ok_or(InputPairError::NoEqual)?;
+    let key = key.trim().to_owned();
+    if key.is_empty() {
+        return Err(InputPairError::MissingKey);
+    }
+    let val = val.trim().to_owned();
+    Ok((key, val))
+}
+
+/// An error that occurred while parsing an input argument
+#[derive(Clone, Debug)]
+enum InputPairError {
```

Comment:
If that would work with clap, I think we could directly return the strings in the error function above and skip this type.